# Homework 1: Introduction to R and Data Analysis - SOLUTION

**Instructor Solution**

This notebook contains the complete solution for Homework 1.

## Exercise 1: Basic R Operations (25 points)

### 1.1 Variables and Assignment (10 points)

In [1]:
# Create variables
my_age <- 25
favorite_color <- "blue"
likes_statistics <- TRUE

# Display values
print(my_age)
print(favorite_color)
print(likes_statistics)

[1] 25
[1] "blue"
[1] TRUE


### 1.2 Arithmetic Operations (15 points)

In [2]:
# Arithmetic operations
sum_result <- 15 + 27
product_result <- 8 * 9
power_result <- 2^5
remainder_result <- 17 %% 3

# Display results with labels
print(paste("Sum of 15 and 27:", sum_result))
print(paste("Product of 8 and 9:", product_result))
print(paste("2 to the power of 5:", power_result))
print(paste("Remainder of 17 divided by 3:", remainder_result))

[1] "Sum of 15 and 27: 42"
[1] "Product of 8 and 9: 72"
[1] "2 to the power of 5: 32"
[1] "Remainder of 17 divided by 3: 2"


## Exercise 2: Vectors and Data Structures (30 points)

### 2.1 Create Vectors (15 points)

In [3]:
# Create vectors
numeric_vector <- c(1:10)  # or c(1, 2, 3, 4, 5, 6, 7, 8, 9, 10)
fruit_vector <- c("apple", "banana", "orange", "grape", "kiwi")
logical_vector <- c(TRUE, FALSE, TRUE, FALSE, TRUE)

# Display vectors
print(numeric_vector)
print(fruit_vector)
print(logical_vector)

 [1]  1  2  3  4  5  6  7  8  9 10
[1] "apple"  "banana" "orange" "grape"  "kiwi"  
[1]  TRUE FALSE  TRUE FALSE  TRUE


### 2.2 Data Frame Creation (15 points)

In [4]:
# Create data frame
student_data <- data.frame(
  Name = c("John", "Jane", "Mike", "Sarah", "Alex"),
  Age = c(20, 19, 21, 20, 22),
  Grade = c("A", "B", "A", "C", "B")
)

# Display data frame
print(student_data)

# Show structure
str(student_data)

    Name Age Grade
1   John  20     A
2   Jane  19     B
3   Mike  21     A
4  Sarah  20     C
5   Alex  22     B
'data.frame':	5 obs. of  3 variables:
 $ Name : chr  "John" "Jane" "Mike" "Sarah" ...
 $ Age  : num  20 19 21 20 22
 $ Grade: chr  "A" "B" "A" "C" ...


## Exercise 3: Data Import and Analysis (25 points)

### 3.1 Load Required Libraries (5 points)

In [5]:
# Load tidyverse library
library(tidyverse)

── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.4     ✔ readr     2.1.5
✖ forcats   1.0.0     ✔ stringr   1.5.1
✔ ggplot2   3.5.2     ✔ tibble    3.3.0
✔ lubridate 1.9.4     ✔ tidyr     1.3.1
✔ purrr     1.0.4     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (<http://conflicted.r-lib.org/>) to force all conflicts to become errors


### 3.2 Create and Import Data (20 points)

In [6]:
# 1. Create CSV data
sales_csv_data <- "ID,Product,Sales
1,Laptop,1200
2,Mouse,25
3,Keyboard,75
4,Monitor,300
5,Tablet,450"

# Write to CSV file
writeLines(sales_csv_data, "sample_sales.csv")

# 2. Read CSV file
sales_data <- read_csv("sample_sales.csv")

# 3. Display first few rows
print(sales_data)

# 4. Show structure
str(sales_data)

# 5. Summary statistics
summary(sales_data)

Rows: 5 Columns: 3
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (1): Product
dbl (2): ID, Sales

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


# A tibble: 5 × 3
     ID Product  Sales
  <dbl> <chr>    <dbl>
1     1 Laptop    1200
2     2 Mouse       25
3     3 Keyboard    75
4     4 Monitor    300
5     5 Tablet     450
spc_tbl_ [5 × 3] (S3: spec_tbl_df/tbl_df/tbl/data.frame)
 $ ID     : num [1:5] 1 2 3 4 5
 $ Product: chr [1:5] "Laptop" "Mouse" "Keyboard" "Monitor" ...
 $ Sales  : num [1:5] 1200 25 75 300 450
 - attr(*, "spec")=
  .. cols(
  ..   ID = col_double(),
  ..   Product = col_character(),
  ..   Sales = col_double()
  .. )
 - attr(*, "problems")=<externalptr> 
       ID         Product              Sales       
 Min.   :1.0   Length:5           Min.   :  25.0  
 1st Qu.:2.0   Class :character   1st Qu.:  75.0  
 Median :3.0   Mode  :character   Median : 300.0  
 Mean   :3.0                      Mean   : 410.0  
 3rd Qu.:4.0                      3rd Qu.: 450.0  
 Max.   :5.0                      Max.   :1200.0  


## Exercise 4: Analysis and Interpretation (20 points)

### 4.1 Data Analysis Questions

In [7]:
# 1. Average value
avg_sales <- mean(sales_data$Sales)
print(paste("Average sales:", avg_sales))

# 2. Maximum and minimum
max_sales <- max(sales_data$Sales)
min_sales <- min(sales_data$Sales)
print(paste("Maximum sales:", max_sales))
print(paste("Minimum sales:", min_sales))

# 3. Number of observations
num_obs <- nrow(sales_data)
print(paste("Number of observations:", num_obs))

[1] "Average sales: 410"
[1] "Maximum sales: 1200"
[1] "Minimum sales: 25"
[1] "Number of observations: 5"


**Interpretation:**

The sales data shows a wide range of product values, from $25 (Mouse) to $1,200 (Laptop). The average sale value is $410, which is influenced by the high-value laptop sale. The data represents 5 different technology products with varying price points. The laptop represents the highest-value item and significantly impacts the mean, while accessories like the mouse represent lower-value transactions. This type of data is typical in retail technology sales where there's a mix of high-value items and accessories.

## Reflection Questions - Sample Answers

1. **What was the most challenging part of this assignment?**
   Understanding the different data types in R and how they behave differently was initially challenging. Also, remembering the correct syntax for creating data frames and using functions like `read_csv()`.

2. **What did you learn about R that you didn't know before?**
   I learned that R has very powerful built-in functions for data analysis, and that the assignment operator `<-` is preferred over `=`. I also discovered how easy it is to create and manipulate vectors and data frames.

3. **How might you use these R skills in future data analysis projects?**
   These basic skills form the foundation for more complex data analysis. I can use vector operations for calculations, data frames for organizing datasets, and the import functions to work with real-world data files. These skills will be essential for statistical analysis and data visualization projects.